Blog/include/functions.php

```diff
@@ -44,16 +44,38 @@ function getComments($blogPostID){ //colon is a security prevention
 		WHERE blogPostID = :blogPostID
 	",
 	array('blogPostID' => $blogPostID)
-);
+)->fetchAll();
 
-	return $result->fetchAll();
+	return $result;
 
 }
 
 function printHeader(){
 		echo "<html>";
 };
 
+@$userID = $_REQUEST['userID'];
```

Comment:
I realize this PR is old and we talked about this today, but just to confirm what we discussed: since this is a file in the /include folder, there shouldn't be any code outside of functions.

For the @ sign, you'll want to do it like this:
```
$userID = @$_REQUEST['userID'];
```
What that's doing is setting a new variable called `$userID` equal to whatever is in the request. But it's possible that `userID` isn't set in the request, in which case this would normally throw a warning. By putting the @ sign before the part where you reference the possibly missing index, you're telling PHP to ignore that warning, and just set `$userID` to NULL.

I won't comment on this everywhere, but the same is true anywhere you use the @ sign. You'll want it to be right before where you reference an index in array where the index might not actually be defined.